test/Azure.Functions.Cli.Tests/KeyVaultReferencesManagerTests.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Colors.Net;
+using FluentAssertions;
+using NSubstitute;
+using Xunit;
+
+using Azure.Functions.Cli.Common;
+
+namespace Azure.Functions.Cli.Tests
+{
+    public class KeyVaultReferencesManagerTests
+    {
+        private IDictionary<string, string> _settings;
+
+        private readonly KeyVaultReferencesManager _keyVaultReferencesManager = new KeyVaultReferencesManager();
+
+        [Theory]
+        [InlineData("", null)]
+        [InlineData("testKey", null)]
+        [InlineData("testKey", "")]
+        [InlineData("testKey", "testValue")]
+        [InlineData("testKey", "testValue: {\"nestedKey\": \"nestedValue\"}")]
+        public void ResolveKeyVaultReferencesDoesNotThrow(string key, string value)
+        {
+            _settings = new Dictionary<string, string>
+            {
+                { Constants.AzureWebJobsStorage, "UseDevelopmentStorage=true" },
+            };
+            _settings.Add(key, value);
+            Exception exception = null;
+            try
+            {
+                _keyVaultReferencesManager.ResolveKeyVaultReferences(_settings);
+            }
+            catch (Exception e)
+            {
+                exception = e;
+            }
+            exception.Should().BeNull();
+        }
+
+        [Theory]
+        [InlineData("test", null, false)]
+        [InlineData("test", "@Microsoft.KeyVault()", true)]
+        [InlineData("test", "@Microsoft.KeyVault(string)", true)]
+        // Attempted Key Vault references are seen as those matching the regular expression
+        // "^@Microsoft.KeyVault(\S*)$".
```

Comment:
```suggestion
        [InlineData("test", "@Microsoft.KeyVault(string)", true)]
        [InlineData("test", "@Microsoft.KeyVault(SecretUri=bad uri)", true)]
        [InlineData("test", "@Microsoft.KeyVault(VaultName=vault;)", true)] // missing secret name
        [InlineData("test", "@Microsoft.KeyVault(SecretName=vault;)", true)] // missing vault name
        [InlineData("test", "@Microsoft-KeyVault()", false)] // hyphen instead of dot
        // Attempted Key Vault references are seen as those matching the regular expression
        // "^@Microsoft.KeyVault(.*)$".
```